packages/backend/src/clients/Aws/S3CacheClient.ts

```diff
@@ -91,11 +96,17 @@ export class S3CacheClient {
                     ContentType: 'application/json',
                     Metadata: sanitizedMetadata,
                 });
-                const response = await this.s3.send(command);
+                await this.s3.send(command);
             } catch (error) {
-                Logger.error(`Failed to upload results to s3. ${error}`);
-                Sentry.captureException(error);
-                throw error;
+                const s3Error = new S3Error(
+                    `Failed to upload results to s3. ${getErrorMessage(error)}`,
+                    {
+                        key,
+                    },
+                );
+                Logger.error(s3Error.message);
+                Sentry.captureException(s3Error);
+                throw s3Error;
```

Comment:
original s3 exception could have some helpful data in it, might be worth logging that one instead of `S3Error` wrapper

Reply:
You can use `S3ServiceException` type from aws package to check if `error instanceof S3ServiceException` and capture it's [`name` property](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/the-response-object.html#response-error-property)